src/governance.ts

```diff
@@ -128,8 +134,54 @@ export class GovernanceCanister {
     return this.responseConverters.toListProposalsResponse(rawResponse);
   };
 
+  public createNeuron = async ({
+    stake,
+    principal,
+    ledgerCanister,
+  }: {
+    stake: ICP;
+    principal: Principal;
+    ledgerCanister: LedgerCanister;
+  }): Promise<void> => {
+    if (stake.toE8s() < E8S_PER_ICP) {
+      throw "Need a minimum of 1 ICP to stake a neuron";
+    }
```

Comment:
I am personally not a big fan of throwing exceptions and would prefer if we return errors as we do with the `LedgerCanister`, but I am not sure which paradigm is better for front-end development. @peterpeterparker maybe you can chime in here?

Reply:
I don't have a strong opinion, but I think that we should be consistent.

Reply:
In this case, I'd prefer we don't throw string errors and instead return typed errors. That's clearer to the caller and is also consistent with the `LedgerCanister`. The method would return something like `Promise<NeuronId | StakeNeuronError>`. See `TransferError` for an example.

Reply:
How do you want the client to check whether the response is correct or not? I can't use `instanceof` nor `typeof` right?

When the response is something like NeuronId I can check for it, but if the good response is also an object, how would I differentiate?

Could we make the TransferError a class and extend the rest of the errors from it?

This way we could check the response with `response instanceof TransferError`.

No need to implement in this PR, but do you think we could consider that or is there a better way?